### PR TITLE
[LBSD-2756] Resolves sharing of post on social media

### PR DESCRIPTION
**Pull request by BrianMwangi21**
Resolves [LBSD-2756] 

When a post is shared and the link opened, it shows the post shared first, with previous posts following and a counter for new updates - if they exist.

[LBSD-2756]: https://sofab.atlassian.net/browse/LBSD-2756?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by BrianMwangi21** [comment]
Screenshot of shared post appearing as the first and the update modal with count above it
image

File: server/liveblog/themes/themes_assets/default/js/theme/view.js [hunk 1]
```diff
@@ -147,17 +147,27 @@ function addPosts(posts, position) {
...
+  }
+
   if (pending.length === 1) {
-    one.toggle('mod--displaynone', false);
-    updates.toggle('mod--displaynone', true);
+    updateToggles(false, true, true);
   } else if (pending.length > 1) {
-    one.toggle('mod--displaynone', true);
-    updates.toggle('mod--displaynone', false);
+    if(permalink) {
```
Comment (eos87) [inline server/liveblog/themes/themes_assets/default/js/theme/view.js:163]: nit:
```suggestion
    if (permalink) {
```

File: server/liveblog/themes/themes_assets/default/js/theme/viewmodel.js [hunk 2]
```diff
@@ -268,15 +272,37 @@ vm.init = function() {
...
+    .catch(error => console.log(error));
   }
 
   var isBlogOpen = LB.blog.blog_status === "open";
   var tenSeconds = 10 * 1000;
 
   if (isBlogOpen) {
-    // let's hit backend right away after load and render latest updates
-    fetchLatestAndRender();
+    if(permalink._id) {
```
Comment (eos87) [inline server/liveblog/themes/themes_assets/default/js/theme/viewmodel.js:292]: nit:
```suggestion
    if (permalink._id) {
```

File: server/liveblog/themes/themes_assets/default/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "liveblog-default-theme",
-  "version": "3.6.68",
+  "version": "3.6.77",
```
Comment (eos87) [inline server/liveblog/themes/themes_assets/default/package.json:3]: I think once this functionality is fully ready we should bump this to be a minor version, like `3.7.0`, it's ok for now though

File: server/liveblog/themes/themes_assets/default/theme.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
     "label": "Liveblog 3 Default SEO Theme",
     "name": "default",
-    "version": "3.6.68",
+    "version": "3.6.77",
```
Comment (eos87) [inline server/liveblog/themes/themes_assets/default/theme.json:4]: Same as my previous comment :)

**Review by eos87 — CHANGES_REQUESTED** [review]
It seems like the references to the build files are broken and not sync between the manifest and the theme file. See my screenshots

This is how it looks when I open a blog in the embed endpoint:
image

and here you have the manifest and theme json files:
image

image

I'm not fully sure why this is happening but perhaps you're using `gulp-watch` without `gulp`? Please double check it. You will need to fix these references manually in both files to point to the right build and then bundle again with gulp maybe.

File: server/liveblog/themes/themes_assets/default/js/theme/view.js (hunk 1 above)
Comment (BrianMwangi21) [reply server/liveblog/themes/themes_assets/default/js/theme/view.js]: Resolved.

File: server/liveblog/themes/themes_assets/default/js/theme/viewmodel.js (hunk 2 above)
Comment (BrianMwangi21) [reply server/liveblog/themes/themes_assets/default/js/theme/viewmodel.js]: Resolved.

**Comment by BrianMwangi21** [comment]
Style and scripts now match in rev-manifest and theme.json

**Review by BrianMwangi21 — COMMENTED** [review]
Requesting review for new changes,

File: server/liveblog/themes/themes_assets/default/js/theme/viewmodel.js [hunk 5]
```diff
@@ -262,24 +310,21 @@ vm.init = function() {
...
-      tags: selectedTags
-    })
-    .then(view.renderPosts)
-    .then(view.initGdprConsentAndRefreshAds);
-  }
-
   var isBlogOpen = LB.blog.blog_status === "open";
   var tenSeconds = 10 * 1000;
 
   if (isBlogOpen) {
```
Comment (eos87) [inline server/liveblog/themes/themes_assets/default/js/theme/viewmodel.js:316]: I think the sharing logic should work also if the blog is no longer open (archived).

File: client/app/scripts/liveblog-embed-script/sharing.ts [hunk 6]
```diff
@@ -1,71 +1,43 @@
-import modalHtml from './ui/shared-post-modal.html';
```
Comment (eos87) [inline client/app/scripts/liveblog-embed-script/sharing.ts:1]: please also delete the `./ui/shared-post-modal.html` template

**Review by eos87 — CHANGES_REQUESTED** [review]
After clicking in the alert "X new updates" the reddit embedded post does not render. Here it is an example below: 
image

Also, when you clicked a shared post and then, instead of clicking the "X new updates" alert you change the order, all the posts are shown but, the alert did not disappear:
image

I left a few comments, please take a look. We're getting there! 

Thanks!

File: server/liveblog/themes/themes_assets/default/theme.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
     "label": "Liveblog 3 Default SEO Theme",
     "name": "default",
-    "version": "3.6.68",
+    "version": "3.6.187",
```
Comment (eos87) [inline server/liveblog/themes/themes_assets/default/theme.json:4]: This version should match the one in package.json file

**Review by eos87 — CHANGES_REQUESTED** [review]
There is one last comment I left regarding version. Please check it 🙏

**Review by eos87 — APPROVED** [review]
all good, well done!

**Pull request merged**